hdfs/src/main/scala/akka/stream/alpakka/hdfs/impl/HdfsFlowStage.scala

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.hdfs.impl
+
+import akka.NotUsed
+import akka.event.Logging
+import akka.stream.alpakka.hdfs._
+import akka.stream.alpakka.hdfs.impl.HdfsFlowLogic.{FlowState, FlowStep, LogicState}
+import akka.stream.alpakka.hdfs.impl.strategy.DefaultRotationStrategy.TimeRotationStrategy
+import akka.stream.alpakka.hdfs.impl.writer.HdfsWriter
+import akka.stream.stage._
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+import cats.data.State
+
+/**
+ * Internal API
+ */
+private[hdfs] final class HdfsFlowStage[W, I, C](
```

Comment:
Please add even `@akka.annotation.InternalApi` as `private[hdfs]` doesn't protect from Java users using it.